Readme.md

```diff
@@ -222,60 +222,113 @@ if(maybeABasket.isNothing()) {
 ```ts
-ap<A, B>(this: Maybe<(v: A) => B>, v: Maybe<A>): Maybe<B>;
-ap<A, B>(this: Maybe<nothing>, v: Maybe<A>): Maybe<nothing>;
-ap<A, B>(this: Maybe<(v: A) => B>, v: Maybe<nothing>): Maybe<nothing>;
+// Maybe<A => B>.ap(other: Maybe<A>): Maybe<B>;
+Maybe.of(v => v * 2).ap(Maybe.of(5));
+-> Maybe<10>;
+// Maybe<nothing>.ap(other: Maybe<A>): Maybe<nothing>;
+Maybe.of(null).ap(Maybe.of(5));
+-> Maybe<null>;
+// Maybe<A => B>.ap(other: Maybe<nothing>): Maybe<nothing>;
+Maybe.of(v => v * 2).ap(Maybe.of(null));
+-> Maybe<null>;
 ```
 - maybe.sequence: when the maybe hold an applicative, it will swap the maybe with the applicative. If the maybe hold nothing it will place the maybe inside an apllicative.
 ```ts
-sequence<A>(this: Maybe<Applicative<A>>,
-        of: (v: Value) => Applicative<Value>,
-    ): Applicative<Maybe<A>>;
-sequence<A>(
-    this: Maybe<nothing>,
-    of: (v: Value) => Applicative<Value>,
-): Applicative<Maybe<nothing>>;
+/**
+ * Maybe<Applicative<A>>.sequence(
+ *     of: Value => Applicative<Value>,
+ * ): Applicative<Maybe<A>>;
+ * */
+Maybe.of(Identity.of(5)).sequence(Identity.of);
+    -> Identity<Maybe<5>>;
+/*
+ * Maybe<nothing>.sequence<A>(
+ *     of: Value => Applicative<Value>,
+ * ): Applicative<Maybe<nothing>>;
+ * */
+Maybe.of(null).sequence(Identity.of);
+    -> Identity<Maybe<null>>;
 ```
 - maybe.traverse: map a function returning an applicative and then swap the applicative with the maybe.
 If the maybe hold nothing, an Apllicative<Maybe<nothing>> will be returned
 ```ts
-traverse<A, B>(
-    this: Maybe<Applicative<A>>,
-    of: (v: Value) => Applicative<Value>,
-    fn: (v: A) => Applicative<B>,
-): Applicative<Maybe<B>>;
-sequence<A, B>(
-    this: Maybe<nothing>,
-    of: (v: Value) => Applicative<Value>,
-    fn: (v: A) => Applicative<B>,
-): Applicative<Maybe<nothing>>;
+/*
+ * Maybe<Applicative<A>>.traverse(
+ *     fn: A => Applicative<B>,
+ *     of: Value => Applicative<Value>,
+ * ): Applicative<Maybe<B>>;
+ * */
+Maybe.of(5).traverse(v => Identity.of(v * 2));
+    -> Identity<Maybe<10>>;
+/*
+ * Maybe<nothing>.traverse(
+ *     of: Value => Applicative<Value>,
+ *     fn: A => Applicative<B>,
+ * ): Applicative<Maybe<nothing>>;
+ * */
+Maybe.of(null).traverse(v => Identity.of(v * 2));
+    -> Identity<Maybe<10>>;
 ```
 - maybe.isNothing: return true if the maybe holds nothing (null or undefined)
+
+```ts
+Maybe<A>.isNothing(): false;
+Maybe<nothing>.isNothing(): true;
+```
+
 - maybe.isJust: return true if the maybe holds a value
+
+```ts
+Maybe<A>.isJust(): true;
+Maybe<nothing>.isJust(): false;
+```
 - maybe.getOrElse: return the Maybe value or the given value if the maybe hold nothing.
 
+```ts
+Maybe<A>.getOrElse(defaultValue: B): A;
+Maybe<nothing>.getOrElse(defaultValue: B): B;
+```
+
 # Either
 
 Either is used to handle possible error, if receiving an error it will preserve it, otherwise it will map normally. If any mapped function would trhow an error, the error will be placed in the Either.
```

Comment:
possible errors.
what does map normally means: do you mean its behavior is like a classical map function ?
trhow-> throw